lib/rules/no-extra-boolean-cast.js

```diff
@@ -63,6 +63,53 @@ module.exports = function(context) {
                     grandparent.callee.name === "Boolean")) {
                 context.report(node, "Redundant double negation in Boolean constructor call.");
             }
+        },
+        "CallExpression": function(node) {
+            var parent = node.parent;
+
+            if (node.callee.type !== "Identifier" || node.callee.name !== "Boolean") {
+                return;
+            }
+
+            // if (<bool>) ...
+            if (parent.type === "IfStatement") {
```

Comment:
How should the conditions be collapsed? Two options I see are to convert to a switch statement (to avoid repeating the `parent.type` part of the comparisons) and to separate the message-choosing logic from the `context.report` call.

Current approach was chosen to match the `UnaryExpression` version of the rule above, but I can update both if another approach is preferred.